alpha/scripts/analyzeEmailFieldForPartner.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+require_once(__DIR__ . '/bootstrap.php');
+
+// parse the command line
+if($argc<2)
+{
+	die("Usage: php $argv[0] <partner id>");
+}
+
+$partnerId = $argv[1];
+$userType = $argv[2];
+$dryRun = $argv[3];
+var_dump($partnerId, $userType, $dryRun);
+
+if (!PartnerPeer::retrieveByPK($partnerId))
+{
+	die("Please enter a valid partner Id!\n");
+}
+
+if (!$userType || ($userType != "admin" && $userType != "user"))
+{
+	die("Please specify if looking for admin users (admin) or regular users (user)\n");
+}
+
+if (!$dryRun || ($dryRun != "dryRun" && $dryRun != "wetRun"))
+{
+	die("Please specify the desired mode of operation: dryRun or wetRun\n");
+}
+
+
+
+function countUsers($partnerId, $isAdmin, $hasEmail = null, $email = null)
+{
+	$emailCriteria = new Criteria();
+	$emailCriteria->add(kuserPeer::PARTNER_ID, $partnerId, Criteria::EQUAL);
+	$emailCriteria->add(kuserPeer::STATUS, KuserStatus::ACTIVE);
+	$emailCriteria->add(kuserPeer::IS_ADMIN, $isAdmin);
+	if($hasEmail == Criteria::ISNULL)
+	{
+		$emailCriteria->add(kuserPeer::EMAIL, null, Criteria::ISNULL);
+	}
+	else if ($hasEmail == Criteria::ISNOTNULL)
+	{
+		if ($email)
+		{
+			$emailCriteria->add(kuserPeer::EMAIL, $email, Criteria::EQUAL);
+		}
+		else
+		{
+			$emailCriteria->add(kuserPeer::EMAIL, null, Criteria::ISNOTNULL);
+		}
+	}
+
+	return kuserPeer::doCount($emailCriteria);
+}
+
+function noEmailPercentage($noEmailUsersCount, $allUsersCount)
+{
+	return (int)(($noEmailUsersCount * 100) /$allUsersCount);
+}
+
+function getUsersWithEmail($partnerId, $isAdmin)
+{
+	$emailCriteria = new Criteria();
+	$emailCriteria->add(kuserPeer::PARTNER_ID, $partnerId, Criteria::EQUAL);
+	$emailCriteria->add(kuserPeer::STATUS, KuserStatus::ACTIVE);
+	$emailCriteria->add(kuserPeer::IS_ADMIN, $isAdmin);
+	$emailCriteria->add(kuserPeer::EMAIL, null, Criteria::ISNOTNULL);
+
+	return kuserPeer::doSelect($emailCriteria);
+}
+
+function countUsersWithDuplicatedEmail($partnerId, $isAdmin)
+{
+	$duplicatedUserCounter = 0;
+	$usersWithEmail = getUsersWithEmail($partnerId, $isAdmin);
+	/* @var $user kuser */
+	foreach ($usersWithEmail as $key => $user)
+	{
+		$usersWithSameEmail = countUsers($partnerId, $isAdmin, Criteria::ISNOTNULL, $user->getEmail());
```

Comment:
if you already returned all the users why not just iterating on them instead of running many of db queries?
also we wiil want the users themselfs, with id and all. we dont have much to do with just counter